Unify Academic and Campuswide Events Row Entries

L1nTsai opened the pull request.

justinwongeecs commented on berkeley-mobile/Events/Academic/AcademicCalendarView.swift:
If you take a look at `AcademicCalendarView.swift` and `CampuswideEventsView.swift` they basically look the same in terms of UI and logic. 

Maybe you can just have one generic view where we can pass in the appropriate parameters in `EventsView.swift`?

Chhumbucket reviewed (changes requested):
Looks good overall just finish Justin's fixes

L1nTsai commented:
- Renamed campus/academic files to EventDetailView.swift and EventRowView.swift
- Consolidated AcademicCalendarView.swift and CampuswideEventsView.swift into EventsView.swift
- Changed the .padding(.horizontal) in EventsDateSectionView.swift to .padding()

Chhumbucket reviewed (approved):
Looks good only nitpick is on .opacity(0) but I do not know if swift has a .hidden modifier

justinwongeecs reviewed (approved):
lgtm!
